src/RadixDlt.NetworkGateway.GatewayApi/Services/IPreviewService.cs

```diff
@@ -113,39 +113,47 @@ public async Task<object> HandlePreviewRequest(JToken request, CancellationToken
         }
     }
 
-    private async Task<object> HandlePreviewAndCreateResponse(JToken request, CancellationToken token)
+    private async Task<GatewayModel.TransactionPreviewResponse> HandlePreviewAndCreateResponse(GatewayModel.TransactionPreviewRequest request, CancellationToken token)
     {
-        CoreModel.TransactionPreviewRequest? coreApiRequest;
-
-        try
-        {
-            if (request is JObject requestObject)
-            {
-                requestObject["network"] = _coreApiHandler.GetNetworkName();
-                coreApiRequest = requestObject.ToObject<CoreModel.TransactionPreviewRequest>();
-            }
-            else
-            {
-                throw InvalidRequestException.FromOtherError("Expected JSON object.");
-            }
-        }
-        catch (JsonSerializationException ex)
+        CoreModel.PublicKey ToCoreModel(GatewayModel.PublicKey publicKey) => publicKey switch
         {
-            throw InvalidRequestException.FromOtherError("Invalid request: " + ex.Message);
-        }
+            GatewayModel.PublicKeyEcdsaSecp256k1 ecdsaSecp256K1 => new CoreModel.EcdsaSecp256k1PublicKey(ecdsaSecp256K1.KeyHex),
+            GatewayModel.PublicKeyEddsaEd25519 eddsaEd25519 => new CoreModel.EddsaEd25519PublicKey(eddsaEd25519.KeyHex),
+            _ => throw new UnreachableException($"Didn't expect {publicKey.GetType().Name} type"),
+        };
 
-        if (coreApiRequest == null)
-        {
-            throw InvalidRequestException.FromOtherError("Expected JSON object.");
-        }
+        var coreRequestFlags = new CoreModel.TransactionPreviewRequestFlags(
```

Comment:
what's the reason that we're not treating `TransactionPreviewRequest` and `TransactionPreviewResponse` as blackbox where we only append network name? it'll be hard to keep it in sync.